Sources/libhostmgr/BuildkiteScriptBuilder.swift

```diff
@@ -0,0 +1,165 @@
+import Foundation
+import TSCBasic
+
+public struct BuildkiteScriptBuilder {
+
+    /// A list of files to run `source` against prior to executing the rest of the script.
+    var dependencies = [String]()
+
+    /// A list of key/value pairs representing environment variables that should be defined at the start of the script.
+    var environmentVariables: [String: Value] = [:]
+
+    /// A list of commands to run in the script.
+    var commands = [Command]()
+
+    public init() {
+        self.environmentVariables["BUILDKITE"] = Value(wrapping: "true")
```

Comment:
As I mentioned in #29, I'm not a fan of having this value defined here, separate to all other env vars which are defined by the caller. Having an exception of defining this one in the `init` here makes for multiple source of truth to look at and search (both in `BuildkiteScriptBuilder` and in its call sites) when trying to make sense of what list of env vars will be included in the final scripts.

I suggest to remove that line from this `init` here, and move it to `GenerateBuildkiteJobScript.swift` instead, to be alongside all the other ad-hoc cases we have for specific env vars like `BUILDKITE_AGENT_NAME` or `CI`.

Reply:
Just to add my two cents, I think it's okay to leave this `BUILDKITE` variable here. This builder is used to generate a buildkite script, any kind of buildkite script, which can be a script to run a job—used by `GenerateBuildkiteJobScript` command, a script to annotate a job, or anything that `buildkite-agent` cli can do. Maybe it's necessary to pass a "BUILD_KITE=true" env to all  `buildkite-agent` calls? I don't have a strong opinion on this though, I'll leave @jkmassel to decide whether to move it or not 😄

Reply:
I might be missing context here but:
 -  I don't see this script builder class, which live in this `hostmgr` repo and tool only, be used from inside the VMs to create scripts for pipelines (like to annotate a build on a pipeline or download artefacts)
 - Even if it were the case despite those n it living in the same infra level,  don't see why those potential scripts to annotate a build or anything that `buildkite-agent` can do would need `BUILDKITE=true` defined to be able to run and work properly
 - In pipelines where we call such `buildkite-agent annotate` subcommands or the like, those commands run by the pipeline are already run in the context of the macOS VM where those env vars are already defined globally (thanks to that very `GenerateBuildkiteJobScript` exporting them already to make them available.

So especially I don't see this `BuildkiteScriptGenerator` being used to generate [pipeline scripts like this one](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/.buildkite/commands/create-github-release.sh#L8) at all, they're just at the same level in the infra.

---

Finally, if you take a step back and look at the `BuildkiteScriptGenerator` as a whole, you'll realize that nothing from its code is specific to Buildkite: the logic of the class could be used to generate any sh or bash script which would contain some `source` commands, some `export` directives and some arbitrary commands at the end, be it for Buildkite or something else. The _only_ think that makes this whole class Buildkite-specific and tying this to it being for Buildkite is this hard-coded `setEnvironmentVariable("BUILDKITE", true)` in its `init`

Reply:
> you'll realize that nothing from its code is specific to Buildkite

That's true, it's a pretty generic "shell script builder".

Reply:
Guess you might consider addressing this in a subsequent PR? 🤔  Or are we keeping the two source of truths here for the time being?

Reply:
🤦‍♂️ Sorry I forgot about this issue. I'll create a PR for it.